solr/solrj/src/java/org/apache/solr/common/cloud/ReplicaCount.java

```diff
@@ -100,6 +166,10 @@ public int total() {
     return countByType.values().stream().reduce(Integer::sum).orElse(0);
   }
 
+  public int count(EnumSet<Replica.Type> replicaTypes) {
```

Comment:
The rationale was the same as for moving `validate` out of here and not having `leaderEligibleTypes` on `Replica.Type`: I would like solrj to have as little business logic as possible (even if it makes the code a bit less elegant in solr-core).

Ideally, solrj and solr-core would have different objects to represent replica types, but that is a much larger refactoring.

Reply:
Until we get to that point (of real separation of API vs implementation) I'd favor code simplicity and readability because I see no real practical impact of keeping SolrJ lean while I do see drawbacks of having more complex code overall.
Would like others to chime in here.
@HoustonPutman @dsmiley for example, what do you think?

Reply:
Agree with @murblanc .  This is very light logic that deals directly with the types at hand.